android/koin-android/src/main/java/org/koin/androidx/viewmodel/factory/StateViewModelFactory.kt

```diff
@@ -13,6 +13,11 @@ class StateViewModelFactory<T : ViewModel>(
     parameters.state?.invoke()
 ) {
     override fun <T : ViewModel> create(key: String, modelClass: Class<T>, handle: SavedStateHandle): T {
+        parameters.defaultArgs?.let { args ->
+            args.keySet().forEach {
+                handle[it] = args.get(it)
+            }
+        }
```

Comment:
Why do we need to do this here? The usual way to do this is to pass this bundle in the `AbstractSavedStateViewModelFactory` constructor parameter.
I see that we are passing a bundle from `ViewModelParameter` state lambda.
So I wonder: why do these two intersect? When is the `parameters.state` lambda used vs the new `defaultArgs`? If there is a key present in both, then the one in `defaultArgs` will override the other, is that expected?

If they serve the same purpose, shouldn't they be one only field of `ViewModelParameter` instead of two?

Also, even if we want to do things like this, for some reason, I would do this:

```kotlin
) : AbstractSavedStateViewModelFactory(
    parameters.registryOwner ?: error("Can't create SavedStateViewModelFactory without a proper stateRegistryOwner"),
    // 👇 probably deserves its own function, and code is not tested but hopefuly it's enough to understand my point
    parameters.state?.invoke()?.let { state -> 
        if (parameters.defaultArgs != null) state.putAll(paramters.defaultArgs)
        else state
    } ?: parameters.defaultArgs
) {
```

Reply:
Koin injected parameters are not necessarily bundle args. But here the API become to be outdated, and perhaps we don't both side

Reply:
After looking at it again I don't think we would require the new `defaultArgs` here. We can make use of `parameters.state` which is being passed to the `AbstractSavedStateViewModelFactory` constructor.

Reply:
I have tried it in a [sample project](https://github.com/rubenquadros/Koin-Compose-Sample/commit/fda566a7f20c085c7eccbcb28e18e54118fb1804) (commit is a bit messy 😅) and it's working as expected. However, it is with the 3.1.4 version. Will try it with updated code.